### PR TITLE
enh(ci): upgrade github actions

### PR DESCRIPTION
**Pull request by paul-oureib**
## Description

**Fixes** # MON-34440

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/robot-test.yml [hunk 1]
```diff
@@ -113,14 +113,14 @@ jobs:
           fetch-depth: 0
 
       - name: Restore image
-        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/robot-test.yml:116]:
```suggestion
        uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

File: .github/workflows/robot-test.yml [hunk 2]
```diff
@@ -113,14 +113,14 @@ jobs:
...
         with:
           path: /tmp/${{inputs.image}}
           key: ${{inputs.image_test}}
           fail-on-cache-miss: true
 
       - name: Restore packages
-        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```
Comment (sc979) [inline .github/workflows/robot-test.yml:123]:
```suggestion
        uses: actions/cache/restore@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
```

**Review by sc979 — APPROVED** [review]

File: .github/workflows/robot-test.yml [hunk 3]
```diff
@@ -174,7 +174,7 @@ jobs:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
       - name: Download Artifacts
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
```
Comment (kduret) [inline .github/workflows/robot-test.yml:177]: this will not work, see https://github.com/centreon/centreon-collect/pull/1025

Comment (sc979) [reply .github/workflows/robot-test.yml:177]: yes indeed, this should be merged after the 1025

Comment (sc979) [reply .github/workflows/robot-test.yml:177]: there's too many conflict when I try to rebase on "MON-33526-adapt-e2e-tests-upgrade-gha"

**Pull request closed without merging**